src/typescript/azure-openapi-validator/rules/utilities/rules-helper.ts

```diff
@@ -28,15 +30,43 @@ function getMostSuccessfulResponseKey(responses: string[]): string {
 }
 
 function getResponseSchema(response: Object, doc): any {
-  const schema = response['schema'];
+  const schema = response["schema"];
   if (schema === undefined || schema === null) {
     return;
   }
-  if ('$ref' in schema) {
-    const schemaRef = response['schema']['$ref'];
-    const schemaPath: string[] = (<string>schemaRef).split('/');
+  if ("$ref" in schema) {
+    const schemaRef = response["schema"]["$ref"];
+    const schemaPath: string[] = (<string>schemaRef).split("/");
     const schemaProperties = doc.definitions[schemaPath[2]].properties;
     return schemaProperties;
   }
   return schema.properties;
-}
\ No newline at end of file
+}
+
+export function getAllResourceProvidersFromPath(path: string): string[] {
+  const resourceProviderRegex: RegExp = new RegExp(/providers\/([\w\.]+)/, "g");
+  return Array.from(matchAll(path, resourceProviderRegex), m => m[1]);
+}
+
+export function getAllWordsFromPath(path: string): string[] {
+  const wordRegex: RegExp = new RegExp(/([\w\.]+)/, "g");
+  return Array.from(matchAll(path, wordRegex), m => m[1]);
+}
+
+export function resourceProviderMustPascalCase(
+  resourceProvider: string
+): boolean {
+  if (resourceProvider.length === 0) {
+    return false;
+  }
+  const pascalCase: RegExp = new RegExp(`^[A-Z][a-z]+(?:\.[A-Z]+[a-z]+)*$`);
+  return pascalCase.test(resourceProvider);
+}
+
+export function resourceTypeMustCamelCase(resourceType: string): boolean {
```

Comment:
do we support letter in the name?

Reply:
Can you provide an example of this case?